watertap/tools/oli_api/util/watertap_to_oli_helper_functions.py

```diff
@@ -0,0 +1,237 @@
+    periodic_table = read_csv(join(file_path, "periodic_table.csv"))
+
+    components = watertap_name.split("_")
+    elements = findall("[A-Z][a-z]?[0-9]*", components[0])
+    element_counts = {}
+    for element in elements:
+        if len(element) == 1:
+            element_counts[element] = 1
+        elif len(element) == 2 and element.isalpha():
+            element_counts[element] = 1
+        elif len(element) == 2 and not element.isalpha():
+            element_counts[element[:-1]] = int(element[-1])
+        elif len(element) == 3 and element[:-1].isalpha():
+            element_counts[element[:-1]] = int(element[-1])
+        elif len(element) == 3 and not element[:-1].isalpha():
+            element_counts[element[:-2]] = int(element[-2:-1])
+        else:
+            raise IOError(f" Too many characters in {element}.")
+
+        element_location = components[0].find(element)
+        if "[" in components[0]:
+            boundary = (components[0].find("["), components[0].find("]"))
+            coefficient = int(components[0][boundary[1] + 1])
+            if element_location > boundary[0] and element_location < boundary[1]:
+                element_counts[element] *= coefficient
+
+    molar_mass = 0
+    for element in element_counts:
+        atomic_mass = float(
+            periodic_table["AtomicMass"][(periodic_table["Symbol"] == element)].values[
+                0
+            ]
+        )
+        molar_mass += element_counts[element] * atomic_mass
+    return molar_mass
+
+
+def get_oli_names(source: dict):
+    """
+    Updates source dictionary with data to populate MCAS property model.
+
+    :param source: dictionary containing WaterTAP names as keys
+
+    :return source: dictionary with OLIName named tuples as keys
+    """
+
+    source = dict(
+        map(lambda k, v: (watertap_to_oli(k), v), source.keys(), source.values())
+    )
+    return source
+
+
+def oli_reverse_lookup(oli_name: str, names_db) -> OLIName:
+    """
+    Looks up WaterTAP formatted name for solute in OLI format, if listed in names_db dictionary.
+
+    :param oli_name: string name of a solute in OLI format
+
+    :return watertap_name: string name of a solute in WaterTAP format
+    """
+
+    if oli_name in names_db:
+        return names_db[oli_name]
+    else:
+        raise IOError(
+            f" Component {oli_name} not found in names_db."
+            + " Update this dictionary to hard code additional OLI names."
+        )
+
+
+"""
+Here follows a dictionary of OLI names and their WaterTAP counterparts.
+
+It functions to aid reverse lookup, i.e., if a name is already in OLI format,
+the necessary data can still be extracted.
+
+TODO: method to add novel (valid) names to names_db
+"""
+
+names_db = {
+    "NAION": "Na_+",
+    "CLION": "Cl_-",
+    "SO4ION": "SO4_2-",
+    "MGION": "Mg_2+",
+    "CAION": "Ca_2+",
+    "KION": "K_+",
+    "HCO3ION": "HCO3_-",
+    "NA2CO3": "Na2CO3",
+    "CO2": "CO2",
+    "H2O": "H2O",
+}
```

Comment:
SHould have a test that verifies the state of the dict that would fail after any updates unless the test was modified too, or at least something that verifies the latest contents of the `names_db`.